src/core/control/xojfile/LoadHandler.cpp

```diff
@@ -1,389 +1,243 @@
-            this->fileVersion = atoi(fileversion);
-        }
-        const char* creator = LoadHandlerHelper::getAttrib("creator", true, this);
-        if (creator) {
-            this->creator = creator;
+    gpointer data = g_malloc(1024);
+    zip_uint64_t readBytes = 0;
+    while (readBytes < length) {
+        const zip_int64_t read = zip_fread(attachmentFile, data, 1024);
+        if (read == -1) {
+            g_free(data);
+            zip_fclose(attachmentFile);
+            logError(FS(PlaceholderString("Could not open attachment: {1}. Error message: Could not read file") %
+                        filename.u8string().c_str()));
+            return;
         }
 
-        this->pos = PARSER_POS_STARTED;
-    } else if (strcmp(elementName, "MrWriter") == 0) {
-        endRootTag = "MrWriter";
-
-        // Read the document version
-        const char* version = LoadHandlerHelper::getAttrib("version", true, this);
-        if (version) {
-            this->creator = "MrWriter ";
-            this->creator += version;
+        const gboolean writeSuccessful =
+                g_output_stream_write_all(outputStream, data, static_cast<gsize>(read), nullptr, nullptr, nullptr);
+        if (!writeSuccessful) {
+            g_free(data);
+            zip_fclose(attachmentFile);
+            logError(FS(
+                    PlaceholderString("Could not open attachment: {1}. Error message: Could not write temporary file") %
+                    filename.u8string().c_str()));
+            return;
         }
 
-        // Document version 1:
-        // Handle it the same as a Xournal document, and don't allow to overwrite
-        this->fileVersion = 1;
-        this->pos = PARSER_POS_STARTED;
-    } else {
-        error("%s", FC(_F("Unexpected root tag: {1}") % elementName));
+        readBytes += static_cast<zip_uint64_t>(read);
     }
-}
-
-void LoadHandler::parseContents() {
-    if (strcmp(elementName, "page") == 0) {
-        this->pos = PARSER_POS_IN_PAGE;
-
-        double width = LoadHandlerHelper::getAttribDouble("width", this);
-        double height = LoadHandlerHelper::getAttribDouble("height", this);
-
-        this->page = std::make_unique<XojPage>(width, height, /*suppressLayer*/ true);
+    g_free(data);
+    zip_fclose(attachmentFile);
 
-        pages.push_back(this->page);
-    } else if (strcmp(elementName, "audio") == 0) {
-        this->parseAudio();
-    } else if (strcmp(elementName, "title") == 0) {
-        // Ignore this tag, it says nothing...
-    } else if (strcmp(elementName, "preview") == 0) {
-        // Ignore this tag, we don't need a preview
-    } else {
-        g_warning("%s", FC(_F("Unexpected tag in document: \"{1}\"") % elementName));
-    }
+    // Map the filename to the extracted temporary filepath
+    char* tmpPath = g_file_get_path(tmpFile.get());
+    // fs::path does not have a std::hash specialization on all compilers
+    this->audioFiles[filename.string()] = tmpPath;
```

Comment:
Reason for this is this defect report: https://cplusplus.github.io/LWG/issue3657
The support for hash was added retrospective to c++17 and therefore has not been implemented and backported in this revision of the compilers. Can you just create a hash class, not in the std namespace (to avoid ODR) and set it as template parameter? You could also check if a std::hash specialization is existing and add it then to std::, but that's probably not worth the effort. But either way comment this, so we know why this hack was necessary.

Alternatively, you can wait until we switch to c++20. @bhennion what do you think?